.github/workflows/build-and-test.yml

```diff
@@ -103,17 +103,43 @@ jobs:
                     requirements-gui.txt > requirements-all.yml
                   conda env update --file requirements-all.yml
 
+            - name: Conda Environment Listing
+              shell: bash -l {0}
+              run: |
+                  conda list
+                  conda list --export > conda-env.txt
+
+            - name: Download Conda Env Artifact
+              continue-on-error: true
+              uses: dawidd6/action-download-artifact@v2
+              with:
+                  workflow: build-and-test.yml
+                  workflow_conclusion: success
```

Comment:
I would put a comment here like "this gets the artifact from the last workflow run that succeeded" just because the meaning of the `workflow_conclusion` option wasn't super obvious to me